frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts

```diff
@@ -9,72 +9,248 @@ import {
   AreaAndPersonFixtures,
   initializeAreaAndPersonData
 } from '../../dev-api/data-init'
-import { deleteApplication, insertApplications } from '../../dev-api'
+import {
+  deleteApplication,
+  execSimpleApplicationActions,
+  getDecisionsByApplication,
+  insertApplications,
+  runPendingAsyncJobs
+} from '../../dev-api'
 import {
   applicationFixture,
   enduserChildFixtureJari,
   Fixture,
-  EmployeeBuilder
+  daycareFixture
 } from '../../dev-api/fixtures'
 import CitizenDecisionsPage from '../../pages/citizen/citizen-decisions'
+import CitizenDecisionResponsePage from '../../pages/citizen/citizen-decision-response'
 import { format } from 'date-fns'
 
 const citizenHomePage = new CitizenHomePage()
 const citizenDecisionsPage = new CitizenDecisionsPage()
+const citizenDecisionResponsePage = new CitizenDecisionResponsePage()
 
 let applicationId: string
-let employee: EmployeeBuilder
 let fixtures: AreaAndPersonFixtures
 let cleanUp: () => Promise<void>
 
 fixture('Citizen decisions')
   .meta({ type: 'regression', subType: 'citizen-decisions' })
   .before(async () => {
     ;[fixtures, cleanUp] = await initializeAreaAndPersonData()
-
-    employee = await Fixture.employee()
-      .with({ roles: ['SERVICE_WORKER'] })
-      .save()
   })
-  .afterEach(logConsoleMessages)
-  .afterEach(async () => {
+  .afterEach(async (t) => {
+    await logConsoleMessages(t)
     await deleteApplication(applicationId)
   })
   .after(async () => {
     await Fixture.cleanup()
     await cleanUp()
   })
 
-test('Citizen sees her decisions', async (t) => {
+test('Citizen sees her decisions, accepts preschool and rejects preschool daycare', async (t) => {
   const application = applicationFixture(
     fixtures.enduserChildFixtureJari,
-    fixtures.enduserGuardianFixture
+    fixtures.enduserGuardianFixture,
+    undefined,
+    'preschool',
+    null,
+    [daycareFixture.id],
+    true
   )
   applicationId = application.id
   await insertApplications([application])
 
-  const decision = await Fixture.decision()
-    .with({
-      applicationId: application.id,
-      employeeId: employee.data.id,
-      unitId: application.form.apply.preferredUnits[0],
-      startDate: application.form.preferredStartDate
-    })
-    .save()
+  await execSimpleApplicationActions(applicationId, [
+    'move-to-waiting-placement',
+    'create-default-placement-plan',
+    'send-decisions-without-proposal'
+  ])
+  await runPendingAsyncJobs()
+
+  const decisions = await getDecisionsByApplication(applicationId)
+  if (decisions.length !== 2) throw Error('Expected 2 decisions')
+  const preschoolDecisionId = decisions.find((d) => d.type === 'PRESCHOOL')?.id
```

Comment:
If decisionbuilder is used you have the id's and can use those directly

Reply:
yeah that is a trade-off